src/api.js

```diff
@@ -1,23 +1,45 @@
-const api = {
-    doImport (data) {
-        try { 
-            window.application.$store.dispatch('importModel', {
-                clientWidth: document.getElementById('svg-grid').clientWidth,
-                clientHeight: document.getElementById('svg-grid').clientHeight,
-                data: JSON.parse(data)
-            });
-        }catch(err){
-            return false;
-        }
-        return true;
-    },
-
-    doExport () {
-        return window.application.$store.getters['exportData'];
-    },
+window.api = {
+  config: null,
+  setConfigAlreadyRun: false,
+  initAlreadyRun: false,
+  doImport(data) {
+    try {
+      window.application.$store.dispatch('importModel', {
+        clientWidth: document.getElementById('svg-grid').clientWidth,
+        clientHeight: document.getElementById('svg-grid').clientHeight,
+        data: JSON.parse(data),
+      });
+    } catch (err) {
+      return false;
+    }
+    return true;
+  },
+  doExport() {
+    return window.application.$store.getters['exportData'];
+  },
+  setConfig(config) {
+    if (this.setConfigAlreadyRun) {
+      throw new Error('This method can only be run once!');
+    }
+    if (config === undefined) { config = {}; }
+    this.config = Object.assign({
+      units: 'm',
+      showMapDialogOnStart: true,
+      online: true,
+      onChange: function change() { window.versionNumber += 1; },
+    }, config);
 
-    setConfig (config) {
-        console.log(config);
-        return true;
+    this.setConfigAlreadyRun = true;
```

Comment:
Now that setConfig is no longer calling startApp, do we still need to protect against multiple calls to setConfig?